xds/src/main/java/io/grpc/xds/XdsLoadBalancer2.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2019 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.ChannelLogger.ChannelLogLevel;
+import io.grpc.ConnectivityState;
+import io.grpc.ConnectivityStateInfo;
+import io.grpc.LoadBalancer;
+import io.grpc.Status;
+import io.grpc.SynchronizationContext.ScheduledHandle;
+import io.grpc.util.ForwardingLoadBalancerHelper;
+import io.grpc.xds.XdsComms.AdsStreamCallback;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+/**
+ * A {@link LoadBalancer} that uses the XDS protocol.
+ *
+ * <p>This class manages fallback handling. The logic for child policy handling and fallback policy
+ * handling is provided by LookasideLb and FallbackLb.
+ */
+// TODO(zdapeng): migrate name to XdsLoadBlancer
+final class XdsLoadBalancer2 extends LoadBalancer {
+
+  private static final long FALLBACK_TIMEOUT_MS = TimeUnit.SECONDS.toMillis(10); // same as grpclb
+
+  private final Helper helper;
+  private final LoadBalancer lookasideLb;
+  private final LoadBalancer.Factory fallbackLbFactory;
+  private final AdsStreamCallback adsCallback = new AdsStreamCallback() {
+    @Override
+    public void onWorking() {
+      if (childPolicyHasBeenReady) {
```

Comment:
It's better to be consistent in terminology. IIUC, this variable is synonyms for "After-Startup", then probably better to name it as `afterStartup`.

Reply:
Oh we could maintain a status for XdsLoadBalancer's working state, such as `STARTUP`, `NORMAL`, `FALLBACK`, if that makes things clearer.

Reply:
For the var names and implementation logic, I tried to keep them the same as the original `XdsLoadBalancer` unless it's obviously necessary, I don't want to fix too much in these restructure PRs.

Actually I was originally using exactly `afterStartup` and changed to `childPolicyHasBeenReady` as suggested. Please see the discussion
https://github.com/grpc/grpc-java/pull/5759#discussion_r286289714